src/directory.c

```diff
@@ -0,0 +1,7 @@
+#include "directory.h"
+
+char	*directory_get_current_dir(void)
+{
+	char buffer[255];
```

Comment:
use heap allocated memory
make value a define / static const

Reply:
@s-t-a-n  In order to do heap allocation I would need to add an extra line : size = pathconf(".", _PC_PATH_MAX);
But I don't think that it is allowed by the subject.

This is how you would use getcwd with allocated memory : 

#include <stdlib.h>
#include <unistd.h>
...
long size;
char *buf;
char *ptr;


size = pathconf(".", _PC_PATH_MAX);


if ((buf = (char *)malloc((size_t)size)) != NULL)
    ptr = getcwd(buf, (size_t)size);
... 

source -> https://pubs.opengroup.org/onlinepubs/007904975/functions/getcwd.html